frontend/app/src/components/defi/LoanInfo.vue

```diff
@@ -0,0 +1,387 @@
+              <div>
+                <amount-display
+                  :value="vault.collateralUsdValue"
+                  fiat-currency="USD"
+                  show-currency="ticker"
+                ></amount-display>
+              </div>
+            </div>
+            <v-divider class="my-4"></v-divider>
+            <div class="d-flex justify-space-between mb-2 font-weight-medium">
+              <div>
+                Current ratio
+              </div>
+              <div>
+                {{
+                  vault.collateralizationRatio
+                    ? vault.collateralizationRatio
+                    : 'NaN'
+                }}
+              </div>
+            </div>
+            <v-btn
+              small
+              rounded
+              block
+              depressed
+              color="grey lighten-3 grey--text text--darken-2"
+              @click="premium ? openWatcherDialog(vault) : ''"
+            >
+              <v-icon x-small left>fa fa-bell-o</v-icon>
+              <span
+                v-if="watchers.length > 0"
+                style="text-transform: none;"
+                class="caption"
+              >
+                Edit {{ watchers.length }} collateralization watcher(s)
+              </span>
+              <span v-else style="text-transform: none;" class="caption">
+                Add collateralization ratio watcher
+              </span>
+              <premium-lock v-if="!premium" size="x-small"></premium-lock>
+            </v-btn>
+          </stat-card>
+          <stat-card class="loan-info__debt mt-5" title="Debt">
+            <div class="d-flex justify-space-between font-weight-medium">
+              <div>
+                Outstanding debt
+              </div>
+              <div>
+                <amount-display
+                  :value="vault.debtValue"
+                  asset="DAI"
+                ></amount-display>
+              </div>
+            </div>
+            <v-divider class="my-4"></v-divider>
+            <div
+              class="loan-info__debt__stability-fee d-flex justify-space-between mb-2 font-weight-medium"
+            >
+              <div>
+                Stability fee
+              </div>
+              <div>
+                {{ vault.stabilityFee }}
+              </div>
+            </div>
+            <div class="d-flex justify-space-between font-weight-medium">
+              <div>
+                Total interest owed
+              </div>
+              <div v-if="premium">
+                <amount-display
+                  v-if="
+                    vault.totalInterestOwed &&
+                    !vault.totalInterestOwed.isNegative()
+                  "
+                  :value="vault.totalInterestOwed"
+                  asset="DAI"
+                ></amount-display>
+                <amount-display
+                  v-else
+                  :loading="typeof vault.totalInterestOwed === 'undefined'"
```

Comment:
doesn't `vault.totalInteresteOwed === undefined` work? it seems weird to do typeof here.

Reply:
I think this is just a pattern I kept from vanilla JS as it works even if the variable hasn't been declared. In typescript I guess it's not necessary.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined
![image](https://user-images.githubusercontent.com/27592/84068844-e133a400-a9c9-11ea-8e93-b38d96af9d9f.png)

Reply:
let's do it the TypeScript way please :)